test/Interop/Cxx/operators/member-inline.swift

```diff
@@ -90,4 +90,55 @@ OperatorsTestSuite.test("DifferentTypesArray.subscript (inline)") {
   expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
 }
 
+OperatorsTestSuite.test("ReadOnlyIntArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyIntArrayByVal(1)
+
+  let result0 = arr[0]
+  let result2 = arr[2]
+  let result4 = arr[4]
+
+  expectEqual(1, result0)
+  expectEqual(3, result2)
+  expectEqual(5, result4)
+
+  arr.setValueAtIndex(42, 2)
+  let result5 = arr[2]
+  expectEqual(42, result5)
+}
+
+OperatorsTestSuite.test("DifferentTypesArrayByVal.subscript (inline)") {
+  var arr = DifferentTypesArrayByVal()
+
+  let resultInt: Int32 = arr[2]
+  let resultDouble: Double = arr[0.1]
+
+  expectEqual(3, resultInt)
+  expectEqual(1.5.rounded(.down), resultDouble.rounded(.down))
+  expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
+}
+
+OperatorsTestSuite.test("ReadOnlyNonTrivialArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyNonTrivialArrayByVal()
+  let NonTrivialByVal = arr[0];
+  if let cStr = NonTrivialByVal.Str {
+    expectEqual("Non-Trivial", String(cString: cStr))
+  }
+
+  expectEqual(1, NonTrivialByVal.a)
+  expectEqual(2, NonTrivialByVal.b)
+  expectEqual(3, NonTrivialByVal.c)
+  expectEqual(4, NonTrivialByVal.d)
+  expectEqual(5, NonTrivialByVal.e)
+  expectEqual(6, NonTrivialByVal.f)
+}
+
+OperatorsTestSuite.test("ReadWritePtrByVal.subscript (inline)") {
```

Comment:
The other thing I'm going to ask is for a few more pointer tests here. That one was of the main topics of this PR (I know it wasn't originally the point, but we spent more time thinking about pointers than anything else by far). These execution tests don't take long to write and easily verify a basic level of correctness.

Reply:
Just a few ideas for types you could test:
 - Returns ref to pointer
 - Const operator returns pointer
 - Returns const pointer
 - Returns pointer to pointer

Reply:
@zoecarver Ah good stuff. I was just about to write some more tests and needed these ideas. Thanks!